gpytorch/variational/whitened_variational_strategy.py

```diff
@@ -0,0 +1,186 @@
+                    induc_induc_covar,
+                    eager_rhss=eager_rhss,
+                    solves=solves,
+                    probe_vectors=probe_vecs,
+                    probe_vector_norms=probe_vec_norms,
+                    probe_vector_solves=probe_vec_solves,
+                    probe_vector_tmats=tmats,
+                )
+
+            # Compute some terms that will be necessary for the predicitve covariance and KL divergence
+            if self.training:
+                interp_data_data_var_plus_mean_diff_inv_quad, logdet = induc_induc_covar.inv_quad_logdet(
+                    torch.cat([induc_data_covar, mean_diff], -1), logdet=True, reduce_inv_quad=False
+                )
+                interp_data_data_var = interp_data_data_var_plus_mean_diff_inv_quad[..., :-1]
+                mean_diff_inv_quad = interp_data_data_var_plus_mean_diff_inv_quad[..., -1]
+            elif beta_features.diagonal_correction.on():
+                interp_data_data_var = induc_induc_covar.inv_quad(induc_data_covar, reduce_inv_quad=False)
+
+            # Compute predictive mean
+            predictive_mean = torch.add(
+                test_mean,
+                induc_induc_covar.inv_matmul(mean_diff, left_tensor=induc_data_covar.transpose(-1, -2)).squeeze(-1),
+            )
+
+            # Compute the predictive covariance
+            if isinstance(variational_dist.lazy_covariance_matrix, RootLazyTensor):
+                predictive_covar = RootLazyTensor(
+                    induc_data_covar.transpose(-1, -2) @ variational_dist.lazy_covariance_matrix.root.evaluate()
+                )
+            else:
+                predictive_covar = MatmulLazyTensor(
+                    induc_data_covar.transpose(-1, -2), predictive_covar @ induc_data_covar
+                )
+
+            if beta_features.diagonal_correction.on():
+                diag_correction = (data_data_covar.diag() - interp_data_data_var).clamp(0, math.inf)
+                predictive_covar = DiagLazyTensor(predictive_covar.diag() + diag_correction)
+
+            # Save the logdet, mean_diff_inv_quad, prior distribution for the ELBO
+            if self.training:
+                self._memoize_cache["prior_distribution_memo"] = MultivariateNormal(induc_mean, induc_induc_covar)
+                self._memoize_cache["logdet_memo"] = -logdet
+                self._memoize_cache["mean_diff_inv_quad_memo"] = mean_diff_inv_quad
+
+            return MultivariateNormal(predictive_mean, predictive_covar)
+
+    def __call__(self, x):
+        if not self.variational_params_initialized.item():
+            prior_dist = self.prior_distribution
+            inv_prior_dist = torch.distributions.MultivariateNormal(
+                prior_dist.mean,
+                prior_dist.lazy_covariance_matrix.add_jitter()
+                .evaluate()
+                .double()
+                .inverse()
+                .type_as(prior_dist.covariance_matrix),
+            )
```

Comment:
See comment above. It would be nice if we didn't have to overwrite `__call__` here.